contracts/tokenLocking/TokenLocking.sol

```diff
@@ -208,11 +203,11 @@ contract TokenLocking is TokenLockingStorage, DSMath { // ignore-swc-123
   // Internal functions
 
   function makeConditionalDeposit(address _token, uint256 _amount, address _user) internal {
-    Lock storage userLock = userLocks[_token][_user];
     if (isTokenUnlocked(_token, _user)) {
+      Lock storage userLock = userLocks[_token][_user];
       userLock.balance = add(userLock.balance, _amount);
     } else {
-      userLock.pendingBalance = add(userLock.pendingBalance, _amount);
+      require(ERC20Extended(_token).transfer(_user, _amount), "colony-token-locking-transfer-failed");
```

Comment:
I'm still sceptical about this being a straight `require`. It opens a window for tokens to mess with any flow that involves transfer of tokens e.g. if the option that's winning in a vote is the one I want, I could lock the token and prevent anyone from revealing any more votes until the reveal period has expired.

Unfortunately, I can't come up with a solution that isn't just `pendingBalance` so I'm not sure how to square this away other than burying our heads in the sand.

Reply:
How so? This is an ERC20 transfer, so it can't fail if the token is locked, since it doesn't know about token locking.

Reply:
Ah, I've not been clear - locking means several things here!

ERC20 only says that this function exists, it makes no guarantees about what it does or any additional functionality. The token itself could implement functionality that makes this function fail (what I was referring to as 'locking' in the above comment). [Our token](https://github.com/JoinColony/colonyToken) is ERC20 compliant, but certainly can stop people transferring with its permission system, for example.

Reply:
Ah, fair. Well, one idea is to hang on to `pendingBalance` for this (rare) scenario, and allow the balance to be tipped back into the main balance whenever someone deposits with `_force`. The require would be turned into an `if` statement.

Reply:
Indeed, that is the only approach that I can think of that works.